### PR TITLE
br, lightning: unify IsScatterRegionFinished

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #51533

Problem Summary:

### What changed and how does it work?

change at BR side: when retry count > 10, random choose and log the first region GetOperator response, instead of every region

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/restore/split.go [hunk 1]
```diff
@@ -481,10 +428,19 @@ func (rs *RegionSplitter) WaitForScatterRegionsTimeout(ctx context.Context, regi
 		reScatterRegions = make([]*split.RegionInfo, 0, len(regionInfos))
 	)
 	for {
-		ctx1 := context.WithValue(ctx, retryTimes, retryCnt)
+		loggedLongRetry := false
 		reScatterRegions = reScatterRegions[:0]
 		for regionID, regionInfo := range leftRegions {
-			ok, rescatter, err := rs.isScatterRegionFinished(ctx1, regionID)
+			if retryCnt > 10 && !loggedLongRetry {
```
Comment (D3Hunter) [inline br/pkg/restore/split.go:434]: maybe use a ratelimiter of burst 1 to log it every xx seconds

**Review by D3Hunter — APPROVED** [review]
rest lgtm

File: br/pkg/restore/split.go (hunk 1 above)
Comment (lance6716) [reply br/pkg/restore/split.go:434]: the old logic is worse... it will print all regions in one retry. And there's a sleep in line 476

**Pull request merged**